lib/job.js

```diff
@@ -157,6 +163,30 @@ class Job extends BaseModel {
             .then(successfulBuilds => successfulBuilds[0]);
     }
 
+    /**
+     * Update a job
+     * @method update
+     * @return {Promise}
+     */
+    update() {
+        // Lazy load factory dependency to prevent circular dependency issues
+        // https://nodejs.org/api/modules.html#modules_cycles
+        /* eslint-disable global-require */
+        const PipelineFactory = require('./pipelineFactory');
+        /* eslint-enable global-require */
+        const pipelineFactory = PipelineFactory.getInstance();
+
+        return super.update()
+            // Update for periodic job
+            .then(newJob => pipelineFactory.get(newJob.pipelineId)
```

Comment:
shouldn't it check whether it's a periodic job prior to invoking `startPeriodic`

Reply:
I was thinking the check should be done in `executor-queue`, since it can simply check if there is a record in redis, but I can also do an extra check here.

Reply:
Yup, I brought up this scenario in Pranav's PR. 
The only accurate way is to check if the old job has periodic annotations OR new job has periodic annotations. But I don't think it's possible since the attribute is already changed once it gets here.